src/java.base/share/conf/security/java.security

```diff
@@ -195,20 +221,116 @@ RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.tls.disabledAlgorithms = \
+    {KeyFactory,EC,*}, \
+    {KeyPairGenerator,EC,*}, \
+    {AlgorithmParameters,EC,*}, \
+    {AlgorithmParameterGenerator,EC,*}, \
+    {Signature,SHA224withECDSA,*}, \
+    {Signature,SHA256withECDSA,*}, \
+    {Signature,SHA384withECDSA,*}, \
+    {Signature,SHA512withECDSA,*}, \
+    {SecureRandom,SHA256DRBG,*}, \
+    {SecureRandom,SHA512DRBG,*}, \
+    {Mac,HmacSHA224,*}, \
+    {Mac,HmacSHA256,*}, \
+    {Mac,HmacSHA384,*}, \
+    {Mac,HmacSHA512,*}, \
+    {Mac,HmacSHA3-224,*}, \
+    {Mac,HmacSHA3-256,*}, \
+    {Mac,HmacSHA3-384,*}, \
+    {Mac,HmacSHA3-512,*}, \
+    {KeyGenerator,kda-hkdf-with-sha224,*}, \
+    {KeyGenerator,kda-hkdf-with-sha256,*}, \
+    {KeyGenerator,kda-hkdf-with-sha384,*}, \
+    {KeyGenerator,kda-hkdf-with-sha512,*}, \
+    {KeyGenerator,SunTls12KeyMaterial,*}, \
+    {KeyGenerator,SunTls12Prf,*}, \
+    {KeyGenerator,SunTls12RsaPremasterSecret,*}, \
+    {KeyGenerator,SunTlsPrf,*}, \
+    {KeyGenerator,SunTlsRsaPremasterSecret,*}, \
+    {KeyGenerator,SunTls12MasterSecret,*}, \
+    {KeyGenerator,SunTlsMasterSecret,*}, \
+    {KeyGenerator,SunTlsKeyMaterial,*}, \
+    {KeyGenerator,HmacSHA224,*}, \
+    {KeyGenerator,HmacSHA256,*}, \
+    {KeyGenerator,HmacSHA384,*}, \
+    {KeyGenerator,HmacSHA512,*}, \
+    {KeyPairGenerator,RSA,*}, \
+    {KeyFactory,RSA,*}, \
+    {AlgorithmParameters,OAEP,*}, \
+    {KeyPairGenerator,RSAPSS,*}, \
+    {KeyFactory,RSAPSS,*}, \
+    {AlgorithmParameters,RSAPSS,*}, \
+    {Signature,SHA224withRSA,*}, \
+    {Signature,SHA256withRSA,*}, \
+    {Signature,SHA384withRSA,*}, \
+    {Signature,SHA512withRSA,*}, \
+    {Signature,RSAPSS,*}, \
+    {KeyAgreement,ECDH,*}, \
+    {MessageDigest,SHA3-224,*}, \
+    {MessageDigest,SHA3-256,*}, \
+    {MessageDigest,SHA3-384,*}, \
+    {MessageDigest,SHA3-512,*}, \
+    {MessageDigest,SHA-224,*}, \
+    {MessageDigest,SHA-256,*}, \
+    {MessageDigest,SHA-384,*}, \
+    {MessageDigest,SHA-512,*}]
+RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.jce.provider.2 = sun.security.provider.Sun [ \
+    {CertificateFactory, X.509, ImplementedIn=Software}, \
     {CertPathBuilder, PKIX, ValidationAlgorithm=RFC5280:ImplementedIn=Software}, \
     {CertPathValidator, PKIX, ValidationAlgorithm=RFC5280:ImplementedIn=Software}, \
     {CertStore, Collection, ImplementedIn=Software}, \
     {CertStore, com.sun.security.IndexedCollection, ImplementedIn=Software}, \
     {Configuration, JavaLoginConfig, *}, \
     {Policy, JavaPolicy, *}]
-RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.jce.provider.3 = SunJSSE
+RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.jce.provider.3 = sun.security.ssl.SunJSSE
+
 RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.javax.net.ssl.keyStore = NONE
 RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.securerandom.provider = OpenJCEPlusFIPS
 RestrictedSecurity.OpenJCEPlusFIPS.FIPS140-3.securerandom.algorithm = SHA512DRBG
+
+#
+# Weakly Enforced RestrictedSecurity mode profile for FIPS 140-3. This profile allows for all algorithms
```

Comment:
I think this profile should consistently use a hyphen:
* Weakly Enforced -> Weakly-Enforced
* FIPS140-3-Weaklyenforced -> FIPS140-3-Weakly-Enforced

Reply:
Done.